src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -0,0 +1,128 @@
+  }
+
+  private parseAppEntries(appEntries: XMLElement[]) {
+    this.witParserService.witnesses$.subscribe(data => this.parsedWitnesses = data);
+    const parsedEntries = appEntries.map((appEntry) => this.parseAppEntry(appEntry));
+
+    return arrayToMap(parsedEntries, 'id');
+  }
+
+  private parseAppEntry(appEntry: XMLElement): ApparatusEntry {
+    this.significantRdgNumber = 0;
+
+    return {
+      type: 'ApparatusEntryComponent',
+      id: appEntry.getAttribute('xml:id') || xpath(appEntry),
+      attributes: this.genericParserService.parseAttributes(appEntry),
+      content: this.parseAppReadings(appEntry),
+      note: this.genericParserService.parse(appEntry.querySelector<XMLElement>(this.noteTagName)),
+      variance: this.getVariance(),
+      originalEncoding: getOuterHTML(appEntry),
+    };
+  }
+
+  private parseAppReadings(appEntry: XMLElement): ApparatusReading[] {
+    return Array.from(appEntry.querySelectorAll(`${this.lemmaTagName}, ${this.readingTagName}`))
+      .map((child: XMLElement) => this.parseAppReading(child));
+  }
+
+  private parseAppReading(rdg: XMLElement): ApparatusReading {
+    return {
+      type: 'ApparatusReadingComponent',
+      tagname: rdg.tagName,
+      id: rdg.getAttribute('xml:id') || xpath(rdg),
+      attributes: this.genericParserService.parseAttributes(rdg),
+      wits: this.parseReadingWitnesses(rdg) || [],
+      content: this.parseAppReadingContent(rdg),
+      significant: this.parseReadingSignificance(rdg),
+    };
+  }
+
+  private parseReadingWitnesses(rdg: XMLElement) {
+    return rdg.getAttribute('wit')?.split('#')
+      .map((el) => removeSpaces(el))
+      .filter((el) => el.length !== 0);
+  }
+
+  private parseAppReadingContent(rdg: XMLElement) {
+    return Array.from(rdg.childNodes)
+      .map((child: XMLElement) => {
+        if (child.nodeName === this.appEntryTagName) {
+          return {
+            type: 'ApparatusEntryComponent',
+            id: child.getAttribute('xml:id') || xpath(child),
+            attributes: {},
+            content: [],
+          };
+        }
+
+        return this.genericParserService.parse(child);
+      });
+  }
+
+  private parseReadingSignificance(rdg: XMLElement) {
+    return rdg.parentElement.tagName === this.readingGroupTagName ?
+      this.isSignificantReading(rdg.parentElement.attributes) :
+      this.isSignificantReading(rdg.attributes);
+  }
+
+  private isSignificantReading(attributes: NamedNodeMap) {
+    const notSignificantReading = AppConfig.evtSettings.edition.notSignificantVariant;
+    let isSignificant = true;
+
+    Array.from(attributes).forEach((attr) => {
+      if (notSignificantReading?.includes(attr.name + '=' + attr.value)) {
+        isSignificant = false;
+      }
+    });
+
+    if (isSignificant) {
+      this.significantRdgNumber++;
+    }
+
+    return isSignificant;
+  }
+
+  private getVariance() {
+    return this.significantRdgNumber / Object.keys(this.parsedWitnesses).length;
```

Comment:
For each apparatus entry, variance is calculated as the number of important reading on the number of witnesses. But if you use this "global" variable (`significantRdgNumber`) this information is kinda lost because you have to go back to the `parseAppEntry` function to see that it is reset to zero. I would probably wait for each app children (lemma and readings) to be parsed and then calculate the variance from that reasults, even if it means to have a double loop over children.

Reply:
could the variance "factor" be a part of the apparatus/reading model?

Reply:
It's already a field of the ApparatusEntry model

Reply:
Not in this file it seems :P